packages/cli/src/commands/new2/new2.ts

```diff
@@ -0,0 +1,281 @@
+  }
+
+  return templates;
+}
+
+async function addToPackages(targetDir: string) {
+  const pkgJson = await fs.readJson(
+    paths.resolveTargetRoot(`${targetDir}/package.json`),
+  );
+  const templateRole = pkgJson.backstage?.role;
+
+  const addToBackend = ['backend-plugin-module', 'backend-plugin'];
+  const addToFrontend = ['frontend-plugin'];
+
+  if (addToBackend.find(role => role === templateRole)) {
+    // 🚨
+  }
+  if (addToFrontend.find(role => role === templateRole)) {
+    // 🚨
+  }
+  return targetDir;
+}
+
+interface Template {
+  description?: string;
+  template: string;
+  type: string;
+  prompts?: ConfigurablePrompt[];
+}
+
+function verifyTemplate(contents: Template, location: string) {
+  if (!contents.template) {
+    throw new Error(
+      'Your CLI template yaml file should have a template path specified',
+    );
+  }
+  const templatePath = paths.resolveTargetRoot(
+    dirname(location),
+    contents.template,
+  );
+
+  if (!fs.existsSync(templatePath)) {
+    throw new Error(
+      `Your CLI template skeleton does not exist: ${templatePath}`,
+    );
+  }
+  if (!['plugin', 'package'].some(type => contents.type === type)) {
+    throw new Error(
+      'Your template must specify a type of either "plugin" or "package"',
+    );
+  }
+  return {
+    prompts: contents.prompts || [],
+    templatePath,
+    templateType: contents.type,
+  };
+}
+
+async function fetchTemplateInformation(template: TemplateLocation) {
+  const { type, target } = template;
+  if (type === 'file') {
+    const location = paths.resolveTargetRoot(target);
+
+    if (!fs.existsSync(location)) {
+      throw new Error(`Not Found: ${location}`);
+    }
+    const contents = parse(fs.readFileSync(location, 'utf-8'));
+    const { prompts, templatePath, templateType } = verifyTemplate(
+      contents,
+      paths.resolveTargetRoot(target),
+    );
+    return {
+      prompts,
+      templatePath,
+      templateType,
+    };
+  } else if (type === 'url') {
+    throw new Error('WIP'); // 🚨 TODO
+  }
+  throw new Error(
+    'Invalid template location type. You must specify either "url" or "file"',
+  );
+}
+
+interface ConfigurablePrompt {
+  id: string;
+  prompt: string;
+  default?: string; // 🚨 TODO: allow numbers/boolean?
+}
+
+async function promptOptions(
+  prompts: ConfigurablePrompt[],
+): Promise<Record<string, string>> {
+  const defaultPrompts = [pluginIdPrompt(), ownerPrompt()];
+  const answers = await inquirer.prompt([
+    ...defaultPrompts,
+    ...prompts.map((prompt: ConfigurablePrompt) => {
+      return {
+        type: 'input',
+        name: prompt.id,
+        message: prompt.prompt,
+        default: prompt.default,
+        validate: (value: string) => {
+          if (!value) {
+            return `Please provide a value for ${prompt.id}`;
+          } else if (!/^[a-z0-9]+(-[a-z0-9]+)*$/.test(value)) {
+            return 'Value must be lowercase and contain only letters, digits, and dashes.';
+          }
```

Comment:
Probably makes sense for us to have a few built-in validation formats if there aren't any in `inquirer` already.